### PR TITLE
feat: Add Switch and Labelled Switch Components

### PR DESCRIPTION
**Pull request by deniaz**

### CODE REVIEW

File: src/components/Switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+export type SwitchProps = {
+    on?: boolean;
+    disabled?: boolean;
+    label?: string;
+    name?: string;
+    size?: Size.Large | Size.Small;
+    onChange?: (e: MouseEvent) => void;
+};
+
+const merge = (classNames: (string | undefined | boolean)[]) => classNames.filter(Boolean).join(" ");
```
Comment (dwirz) [inline src/components/Switch/Switch.tsx:18]: nitty-gritty helper. how about extract it to a utility file and reuse it in the other please we do similar stuff? cc @SamuelAlev

Comment (SamuelAlev) [reply src/components/Switch/Switch.tsx:18]: Really great idea, remind me of [classnames](https://www.npmjs.com/package/classnames)

File: src/components/Switch/Switch.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        cy.get("@Switch").should("have.value", "false");
+
+        cy.get("@Switch").click();
+        cy.get("@Switch").should("have.value", "true");
+    });
+
+    it("should do nothing if disabled", () => {
+        mount(<Component name="switch-test-disabled" disabled />);
+
+        cy.get("[data-test-id=Switch]").as("Switch");
```
Comment (SamuelAlev) [inline src/components/Switch/Switch.spec.tsx:30]: We use kebab-case for those

File: src/components/Switch/Switch.tsx (hunk 1 above)
Comment (deniaz) [reply src/components/Switch/Switch.tsx]: alright, done here: https://github.com/Frontify/arcade/pull/8/commits/cefe40c2b5089d3d87ebf13018faab91ebbfd61c

File: src/components/Switch/Switch.spec.tsx (hunk 2 above)
Comment (deniaz) [reply src/components/Switch/Switch.spec.tsx]: https://github.com/Frontify/arcade/pull/8/commits/7366c57ecc30e14003f3251c1bad7f8a70a5e7fd I adjusted the generator so it's now kebab-case by default ;)

File: src/components/Switch/Switch.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    useEffect(() => {
+        setOn(args.on);
+    }, [args.on]);
+
+    return (
+        <Switch
+            {...args}
+            onChange={() => {
+                toggle();
+            }}
```
Comment (SamuelAlev) [inline src/components/Switch/Switch.stories.tsx:32]:
```suggestion
            onChange={toggle}
```

File: src/components/Switch/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import Label from "../Form/Label/Label";
+import css from "./Switch.module.css";
+
+export type SwitchProps = {
+    on?: boolean;
+    disabled?: boolean;
+    label?: string;
+    name?: string;
+    size?: Size.Large | Size.Small;
+    onChange?: (e: MouseEvent) => void;
```
Comment (SamuelAlev) [inline src/components/Switch/Switch.tsx:16]:
```suggestion
    onChange?: (event: MouseEvent) => void;
```
nit: we decided to not use the shorthand `e` but `event` everywhere

**Review by SamuelAlev — APPROVED** [review]
Except the 2 little nit, works fine!

File: src/components/Switch/Switch.tsx (hunk 4 above)
Comment (deniaz) [reply src/components/Switch/Switch.tsx:16]: @SamuelAlev are things like that documented somewhere like coding guidelines? cc @dwirz

**Review by dwirz — APPROVED** [review]

**Pull request merged**